services/web/server/src/simcore_service_webserver/projects/projects_db.py

```diff
@@ -188,6 +193,7 @@ def __init__(self, app: web.Application):
         # TODO: shall be a weak pointer since it is also contained by app??
         self._app = app
         self._engine = app.get(APP_DB_ENGINE_KEY)
+        self._thread_pool = next(_setup_thread_pool())
```

Comment:
1. next(.) will run until yield but you need yet another next(.) to get out of the context manager, i.e. the pool is not closed here
2. IMO you should simply use the one in ``app[kTHREAD_POOL_EXECUTOR)``. For that you need to make sure the thread is already setup.

This should be done as with the http client. Check the  the **two functions** implemented in [servicelib/client_session.py](https://github.com/ITISFoundation/osparc-simcore/blob/master/packages/service-library/src/servicelib/client_session.py)

- The first guarantees a singe thread-pool instance  app-wide (lazily init by first caller)
- The second guarantees correct setup/teardown of pool in a app scope

Reply:
What about this ``_thread_pool``? I guess you missed it